backend/announcement/announcement-api.js

```diff
@@ -0,0 +1,59 @@
+/* API Endpoints for announcement */
+const mongoose = require('mongoose');
+const MongooseConnector = require('../db-helper');
+
+const confirmValidDate = (date, compDate = Date.now()) => {
+    date = +date;
+    return !!date && (new Date(date) >= compDate);
+};
+
+const checkSuccess = (res, val) => {
+    if (!val) {
+      res.status(500).json({
+        message: 'Database error',
+      });
+      return;
+    }
+  
+    res.status(200).json({
+      successful: val,
+    });
+};
+
+module.exports = (app) => {
+    app.get('/api/announcement', async (req, res) => {
+        const date = req.query.date
+        const author = req.query.author
+        const title = req.query.title
+        if(date !== undefined){
+          let announcements = await MongooseConnector.getAnnouncementByDate();
```

Comment:
Need to pass date to function and it should be just
if (confirmValidDate(date))
if (date !== undefined) allows date to be empty if defined.